torch/csrc/api/include/torch/nn/modules/dropout.h

```diff
@@ -14,66 +14,90 @@ namespace torch {
-/// 2-D features and
-/// [Dropout3d](https://pytorch.org/docs/stable/nn.html#torch.nn.Dropout3d) for
-/// 3-D features. This `FeatureDropout` module can instead deal with both 2-D
-/// and 3-D features.
-class TORCH_API FeatureDropoutImpl
-    : public detail::DropoutImplBase<FeatureDropoutImpl> {
- public:
-  explicit FeatureDropoutImpl(const DropoutOptions& options_ = DropoutOptions());
+/// A `ModuleHolder` subclass for `DropoutImpl`.
+/// See the documentation for `DropoutImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(Dropout);
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Dropout2d ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Applies dropout over a 2-D input.
+/// See https://pytorch.org/docs/master/nn.html#torch.nn.Dropout2d to learn
+/// about the exact behavior of this module.
+class TORCH_API Dropout2dImpl : public detail::_DropoutNd<DropoutImpl> {
+public:
+  using detail::_DropoutNd<DropoutImpl>::_DropoutNd;
 
-  /// During training, applies a noise mask to the input tensor.
-  /// During evaluation, applies an identity function.
-  Tensor forward(const Tensor& input);
+  Tensor forward(Tensor input);
 
-  /// Pretty prints the `FeatureDropout` module into the given `stream`.
+  /// Pretty prints the `Dropout2d` module into the given `stream`.
   void pretty_print(std::ostream& stream) const override;
 };
 
-/// A `ModuleHolder` subclass for `DropoutImpl`.
-/// See the documentation for `DropoutImpl` class to learn what methods it
+/// A `ModuleHolder` subclass for `Dropout2dImpl`.
+/// See the documentation for `Dropout2dImpl` class to learn what methods it
 /// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
 /// module storage semantics.
-TORCH_MODULE(Dropout);
+TORCH_MODULE(Dropout2d);
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Dropout3d ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Applies dropout over a 3-D input.
+/// See https://pytorch.org/docs/master/nn.html#torch.nn.Dropout3d to learn
+/// about the exact behavior of this module.
+class TORCH_API Dropout3dImpl : public detail::_DropoutNd<DropoutImpl> {
+public:
+  using detail::_DropoutNd<DropoutImpl>::_DropoutNd;
+
+  Tensor forward(Tensor input);
+
+  /// Pretty prints the `Dropout3d` module into the given `stream`.
+  void pretty_print(std::ostream& stream) const override;
+};
+
+/// A `ModuleHolder` subclass for `Dropout3dImpl`.
+/// See the documentation for `Dropout3dImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(Dropout3d);
 
-/// A `ModuleHolder` subclass for `FeatureDropoutImpl`.
-/// See the documentation for `FeatureDropoutImpl` class to learn what methods
-/// it provides, or the documentation for `ModuleHolder` to learn about
-/// PyTorch's module storage semantics.
-TORCH_MODULE(FeatureDropout);
```

Comment:
I think we might want to deprecate `FeatureDropout` for one release and then remove it in the next one. For 1.4 (which will be branch cut on 11/20) we can deprecate it first by printing a warning message in the `FeatureDropout` constructor asking people to move to `Dropout2d` / `Dropout3d`, and in the 1.5 release we can remove `FeatureDropout` completely :D